pkg/kobject/kobject.go

```diff
@@ -137,16 +137,16 @@ type ServiceConfig struct {
 	MemReservation    yaml.MemStringorInt `compose:""`
 	DeployMode        string              `compose:""`
 	// DeployLabels mapping to kubernetes labels
-	DeployLabels       map[string]string           `compose:""`
-	DeployUpdateConfig dockerCliTypes.UpdateConfig `compose:""`
-	TmpFs              []string                    `compose:"tmpfs"`
-	Dockerfile         string                      `compose:"dockerfile"`
-	Replicas           int                         `compose:"replicas"`
-	GroupAdd           []int64                     `compose:"group_add"`
-	Volumes            []Volumes                   `compose:""`
+	DeployLabels       map[string]string                    `compose:""`
+	DeployUpdateConfig dockerCliTypes.UpdateConfig          `compose:""`
+	TmpFs              []string                             `compose:"tmpfs"`
+	Dockerfile         string                               `compose:"dockerfile"`
+	Replicas           int                                  `compose:"replicas"`
+	GroupAdd           []int64                              `compose:"group_add"`
+	Volumes            []Volumes                            `compose:""`
 	Secrets            []dockerCliTypes.ServiceSecretConfig
-	HealthChecks       HealthChecks      `compose:""`
-	Placement          map[string]string `compose:""`
+	HealthChecks       HealthChecks                         `compose:""`
+	Placement          []corev1.NodeSelectorRequirement     `compose:""`
```

Comment:
The `ServiceConfig` struct is more similar to docker-compose data structure. By far we haven't introduce any kubernetes-specific fields into it. So i think we should keep it that way for now. You can define a custom struct to represent the placement requirement of the docker-compse and parse it in kubernetes package

Reply:
That's a better way